controller/ingress.go

```diff
@@ -184,6 +185,141 @@ func NewALBIngressFromIngress(ingress *extensions.Ingress, ac *ALBController) (*
+		return nil, false
+	}
+
+	log.Infof("Fetching resource recordset for %s/%s %s", ingressID, namespace, ingressName, hostname)
+	resourceRecordSet, err := awsutil.Route53svc.DescribeResourceRecordSets(zone.Id,
+		&hostname)
+	if err != nil {
+		log.Errorf("Failed to find %s in AWS Route53", ingressID, hostname)
+	}
+
+	rs := &alb.ResourceRecordSet{
+		IngressID: &ingressID,
+		ZoneID:    zone.Id,
+		CurrentResourceRecordSet: resourceRecordSet,
+	}
+
+	lb := &alb.LoadBalancer{
+		ID:                  loadBalancer.LoadBalancerName,
+		IngressID:           &ingressID,
+		Hostname:            aws.String(hostname),
+		CurrentLoadBalancer: loadBalancer,
+		ResourceRecordSet:   rs,
+		CurrentTags:         tags,
+	}
+
+	albIngress := NewALBIngress(namespace, ingressName, clusterName)
+	albIngress.LoadBalancers = []*alb.LoadBalancer{lb}
+
+	targetGroups, err := awsutil.ALBsvc.DescribeTargetGroups(loadBalancer.LoadBalancerArn)
+	if err != nil {
+		log.Fatalf("%s", ingressID, err)
+	}
+
+	for _, targetGroup := range targetGroups {
+		tags, err := awsutil.ALBsvc.DescribeTags(targetGroup.TargetGroupArn)
+		if err != nil {
+			log.Fatalf("%s", ingressID, err)
+		}
+
+		svcName, ok := tags.Get("ServiceName")
+		if !ok {
+			log.Infof("The LoadBalancer %s does not have an Namespace tag, can't import", ingressID, *loadBalancer.LoadBalancerName)
+			return nil, false
+		}
+
+		tg := &alb.TargetGroup{
+			ID:                 targetGroup.TargetGroupName,
+			IngressID:          &ingressID,
+			SvcName:            svcName,
+			CurrentTags:        tags,
+			CurrentTargetGroup: targetGroup,
+		}
+		log.Infof("Fetching Targets for Target Group %s", ingressID, *targetGroup.TargetGroupArn)
+
+		targets, err := awsutil.ALBsvc.DescribeTargetGroupTargets(targetGroup.TargetGroupArn)
+		if err != nil {
+			log.Fatalf("%s", ingressID, err)
+		}
+		tg.CurrentTargets = targets
+		lb.TargetGroups = append(lb.TargetGroups, tg)
+	}
+
+	listeners, err := awsutil.ALBsvc.DescribeListeners(loadBalancer.LoadBalancerArn)
+	if err != nil {
+		log.Fatalf("%s", ingressID, err)
+	}
+
+	for _, listener := range listeners {
+		log.Infof("Fetching Rules for Listener %s", ingressID, *listener.ListenerArn)
+		rules, err := awsutil.ALBsvc.DescribeRules(listener.ListenerArn)
+		if err != nil {
+			log.Fatalf("%s", ingressID, err)
+		}
+
+		l := &alb.Listener{
+			CurrentListener: listener,
+			IngressID:       &ingressID,
+		}
+
+		for _, rule := range rules {
+			var svcName string
+			for _, tg := range lb.TargetGroups {
+				if *rule.Actions[0].TargetGroupArn == *tg.CurrentTargetGroup.TargetGroupArn {
+					svcName = tg.SvcName
+				}
+			}
+
+			log.Debugf("Assembling rule with svc name: %s", ingressID, svcName)
+			l.Rules = append(l.Rules, &alb.Rule{
+				IngressID:   &ingressID,
+				SvcName:     svcName,
+				CurrentRule: rule,
+			})
+		}
+
+		// Set the highest known priority to the amount of current rules plus 1
+		lb.LastRulePriority = int64(len(l.Rules)) + 1
+
```

Comment:
Would be cool to put a `log.Infof` here stating something along the line of:

`[foo-bar]: Ingress rebuilt from existing ALB in AWS`.